### DISCUSSION TITLE
[2021.08.27] 철벽 보안 알고리즘

### DISCUSSION THREAD

**Discussion by pyunji** [pyunji/Hyundai-ITnE-Team_QuadCore#69] [Algorithm]
[철벽 보안 알고리즘](https://www.acmicpc.net/problem/9322)
# 문제 오역으로 인해 직접 원문을 번역해  다시 올립니다.

Alice는 공개키-개인키 쌍으로 보안을 유지하는 시스템이 불편하다고 생각했다.
그래서 그녀는 RSA(Reallly Secure Algorithm) 이라고 불리는 공개키-공개키 암호화 방식을 발명했다. 
이 알고리즘은 아래와 같이 작동한다.

---

- 한 단어는 1개 이상 10개 이하의 알파벳 대문자로 이루어져있다.
- 한 문장은 여러 단어로 이루어져 있고, 단어와 단어 사이에는 공백이 추가되어있다.
- `제 1 공개키`는 중복없는 단어들로 이루어져있다.
- 제 1 공개키에 `암호화 규칙 theta`를 적용하여 `제 2 공개키`를 만들었다.
- 평문(암호화되지 않은 문장)은 공개키와 단어의 개수가 같다.
  - 평문을 이루는 단어들은 중복될 수도 있다.
- **평문에 암호화 규칙 theta를 적용하여 암호문을 만든다.**
---

두 개의 공개키와 하나의 암호문이 주어졌을 때, 암호문을 해독해 평문을 복구해라.

**Comment by hyk339** [comment]
문제 이해가 어렵습니당..ㅠ
![KakaoTalk_20210829_150906761](https://user-images.githubusercontent.com/82932127/131240648-cb95fef2-6a94-4feb-8c33-50f3c1eb91ee.jpg)
![KakaoTalk_20210829_151348170](https://user-images.githubusercontent.com/82932127/131240650-0bd7ed9a-b30a-4b4c-8d04-dd5fb52d13f3.jpg)

**Reply by pyunji** [reply]
제 생각에는 문제가 잘못 번역 된 것 같아요. 
제가 다시 번역해서 올려드릴테니 나중에 다시 풀어주세요

**Comment by pyunji** [comment]
[철벽 보안 알고리즘 solved](https://github.com/pyunji/Hyundai-ITnE-Team_QuadCore/blob/pyunji/pyunji/BOJ/BOJ_9322.java)

**Comment by chaeyeon4u** [comment]
[철벽 보안 알고리즘 solved](https://github.com/pyunji/Hyundai-ITnE-Team_QuadCore/blob/master/ycy/homework/Java/BOJ_9322.java)

**Comment by hyk339** [comment]
[철벽 보안 알고리즘 solved](https://github.com/hyk339/Hyundai-ITnE-Team_QuadCore/blob/master/khy/baekjoon/BOJ_9322.java)

**Comment by leechang4** [comment]
[철벽 보안 알고리즘 solved](https://github.com/pyunji/Hyundai-ITnE-Team_QuadCore/blob/master/lch/BOJ/BOJ%209322.java)
